x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/analyzer/VerifierErrorMessagesTests.java

```diff
@@ -416,65 +440,101 @@ public void testInOnFieldTextWithNoKeyword() {
     }
 
     public void testNotSupportedAggregateOnDate() {
-        assertEquals("1:8: [AVG(date)] argument must be [numeric], found value [date] type [datetime]",
+        assertEquals("1:8: argument of [AVG(date)] must be [numeric], found value [date] type [datetime]",
             error("SELECT AVG(date) FROM test"));
     }
 
-    public void testInvalidTypeForStringFunction_WithOneArg() {
-        assertEquals("1:8: [LENGTH] argument must be [string], found value [1] type [integer]",
+    public void testInvalidTypeForStringFunction_WithOneArgString() {
+        assertEquals("1:8: argument of [LENGTH(1)] must be [string], found value [1] type [integer]",
             error("SELECT LENGTH(1)"));
     }
 
+    public void testInvalidTypeForStringFunction_WithOneArgNumeric() {
+        assertEquals("1:8: argument of [CHAR('foo')] must be [integer], found value ['foo'] type [keyword]",
```

Comment:
Added a test: https://github.com/elastic/elasticsearch/pull/39427/commits/cda66caaf1abb7a0d3afcd427ade3b1912bcd1a2#diff-334bd2f8c33b60789817181fcef4ca52R457

Good thinking! But seems fine since we use `sourceText()` on the offending function and it's not showing the whole expression that can be complex and result in confusion for the user.